tests/func/test_checkout.py

```diff
@@ -850,3 +850,22 @@ def test_checkouts_for_pipeline_tracked_outs(tmp_dir, dvc, scm, run_copy):
 
     (tmp_dir / "ipsum").unlink()
     assert set(dvc.checkout()["added"]) == {"bar", "ipsum"}
+
+
+@pytest.mark.parametrize(
+    "workspace", [pytest.lazy_fixture("s3")], indirect=True
+)
+def test_checkout_external_modified_file(tmp_dir, dvc, scm, mocker, workspace):
+    # regression: happened when file in external output changed and checkout
+    # was attempted without force, dvc checks if it's present in its cache
+    # before asking user to remove it.
+    workspace.gen("foo", "foo")
+    dvc.add(str(workspace / "foo"), external=True)
+    scm.add(["foo.dvc"])
+    scm.commit("add foo")
+
+    workspace.gen("foo", "foobar")  # messing up the external outputs
+    mocker.patch("dvc.prompt.confirm", return_value=True)
+    dvc.checkout()
```

Comment:
Maybe
```suggestion
    dvc.checkout(force=True)
```

Reply:
This bug is only in effect on `force = False`. When checkout is forced, we never check for `already_cached`.

Reply:
Instead, we just remove it, without checking if it's cached.